python_files/generate_review_figures.py

```diff
@@ -0,0 +1,625 @@
+interactions_df = interactions_df.pivot(index='fov', columns='feature', values='value').reset_index()
+interactions_df.to_csv('/Users/csowers/Documents/TONIC_reviews/NT_cell_interactions/Epi_TME_interactions.csv', index=False)
+
+# run SpaceCat with interaction features included
+data_dir = '/Users/csowers/Documents/TONIC_reviews/NT_cell_interactions'
+adata = anndata.read_h5ad(os.path.join(data_dir, 'adata', 'adata.h5ad'))
+
+# read in image level dataframes
+fiber_df = pd.read_csv(os.path.join(data_dir, 'fiber_stats_table.csv'))
+fiber_tile_df = pd.read_csv(os.path.join(data_dir, 'fiber_stats_per_tile.csv'))
+mixing_df = pd.read_csv(os.path.join(data_dir, 'formatted_mixing_scores.csv'))
+kmeans_img_proportions = pd.read_csv(os.path.join(data_dir, 'neighborhood_image_proportions.csv'))
+kmeans_compartment_proportions = pd.read_csv(os.path.join(data_dir, 'neighborhood_compartment_proportions.csv'))
+pixie_ecm = pd.read_csv(os.path.join(data_dir, 'pixie_ecm_stats.csv'))
+ecm_frac = pd.read_csv(os.path.join(data_dir, 'ecm_fraction_stats.csv'))
+ecm_clusters = pd.read_csv(os.path.join(data_dir, 'ecm_cluster_stats.csv'))
+cell_interactions = pd.read_csv(os.path.join(data_dir, 'Epi_TME_interactions.csv'))
+
+# specify cell type pairs to compute a ratio for
+ratio_pairings = [('CD8T', 'CD4T'), ('CD4T', 'Treg'), ('CD8T', 'Treg'), ('CD68_Mac', 'CD163_Mac')]
+# specify addtional per cell and per image stats
+per_cell_stats=[
+    ['morphology', 'cell_cluster', ['area', 'major_axis_length']],
+    ['linear_distance', 'cell_cluster_broad', ['distance_to__B', 'distance_to__Cancer', 'distance_to__Granulocyte', 'distance_to__Mono_Mac',
+                                               'distance_to__NK', 'distance_to__Other', 'distance_to__Structural', 'distance_to__T']]
+
+]
+per_img_stats=[
+    ['fiber', fiber_df],
+    ['fiber', fiber_tile_df],
+    ['mixing_score', mixing_df],
+    ['kmeans_cluster', kmeans_img_proportions],
+    ['kmeans_cluster', kmeans_compartment_proportions],
+    ['pixie_ecm', pixie_ecm],
+    ['ecm_fraction', ecm_frac],
+    ['ecm_cluster', ecm_clusters],
+    ['cell_interactions', cell_interactions]
+]
+
+features = SpaceCat(adata, image_key='fov', seg_label_key='label', cell_area_key='area',
+                    cluster_key=['cell_cluster', 'cell_cluster_broad'],
+                    compartment_key='compartment', compartment_area_key='compartment_area')
+# Generate features and save anndata
+adata_processed = features.run_spacecat(functional_feature_level='cell_cluster', diversity_feature_level='cell_cluster', pixel_radius=50,
+                                        specified_ratios_cluster_key='cell_cluster', specified_ratios=ratio_pairings,
+                                        per_cell_stats=per_cell_stats, per_img_stats=per_img_stats)
+
+adata_processed.write_h5ad(os.path.join(data_dir, 'adata', 'adata_processed.h5ad'))
```

Comment:
Since there's no increase in accuracy, either the features are useless or redundant. To figure out which, can you generate a correlation heatmap similar to Figure 2c? However, instead of being symmetric, this one would have just the NT features as columns, as just the spacecat features as rows. 

If you could do the same thing for the mixing scores (spacecat as rows, mixing scores as columns) that would be great. 

Could you also make a version of each of those where you only keep the rows that have at least one high correlation value, to make it easier to see which features are important?

Reply:
These are on the NAS in `/review_figures/NT_interaction_features` along with the high correlation plots. The interaction features seem to be very similar to features we already include (see `interaction_feature_clustermap_filtered_corr70.pdf` where 9.8% of SpaceCat features have >0.7 correlation with at least one interaction feature).

<img width="500" alt="Screenshot 2024-12-17 at 4 39 05 PM" src="https://github.com/user-attachments/assets/722aa05a-9ceb-4950-a0e2-a6d09609c827" />
<img width="500" alt="Screenshot 2024-12-17 at 4 39 40 PM" src="https://github.com/user-attachments/assets/87becbcf-270d-428c-9766-a38ee0cd4c2f" />